tests/integration/telemetry/stats/prometheus/http/stats.go

```diff
@@ -170,6 +214,10 @@ func buildQuery() (sourceQuery, destinationQuery, appQuery string) {
 		"source_workload_namespace":      ns.Name(),
 	}
 	for k, v := range labels {
+		// These two fields remain unknown so we skip them for VMs
+		if destination == "vm" && (k == "destination_app" || k == "destination_version") {
+			continue
+		}
```

Comment:
I tried to relabel the labels but these two fields are still unknown. Is there anyway to let the prometheus know the app and version info? Maybe by relabeling?

Reply:
app and version should could from the metrics on the proxies themselves. we shouldn't try to relabel. for tests, we can simulate this by adding:

```json
"LABELS": {
  "app": "server",
  "version": "v1"
}
```

to the custom bootstrap information.

Reply:
or by setting the `ISTIO_METAJSON_LABELS` env var.

Reply:
Thanks for the suggestion! But what I am confused of is, why we have to simulate for tests? Won't test query info from envoy as well? From port 15090?

Reply:
envoy gets the info on pod labels from its bootstrap config. typically, they are provided by sidecar injector. for vms, we have to hand supply them (for the moment).

Reply:
Thanks! That makes sense.